File: src/main/java/com/xellitix/commons/net/compat/java/uri/UriFactory.java
package com.xellitix.commons.net.compat.java.uri;

import java.net.URI;
import java.net.URISyntaxException;

/**
 * {@link URI} factory.
 *
 * @author Grayson Kuhns
 */
public interface UriFactory {

  /**
   * Creates a {@link URI}.
   *
   * @param uri The string representation.
   * @return The {@link URI}.
   * @throws URISyntaxException If the string representation is malformed.
   */
  URI create(String uri) throws URISyntaxException;
}

File: src/main/kotlin/com/xellitix/commons/net/uri/UriFactory.kt
package com.xellitix.commons.net.uri

import com.google.inject.Singleton
import java.net.URI
import java.net.URISyntaxException

/**
 * [URI] factory.
 */
@Singleton
class UriFactory {

    /**
     * Creates a [URI].
     */
    fun create(uri: String): URI {
        return try {
            URI(uri)
        } catch (ex: URISyntaxException) {
            throw UriSyntaxException(ex)
        }
    }
}
